### PR TITLE
Fix various issues detected by static analysis and invalid file mode in systemd service file

### PR DESCRIPTION
**Pull request by jrisc**
This PR includes some fixes based on reports from static analysis tools (mainly related to unreleased resources and uninitialized variables), and a fix for the provided systemd service file, which causes invalid file mode to be set on cache folders.

### CODE REVIEW

File: src/client/gpm_accept_sec_context.c [hunk 1]
```diff
@@ -105,6 +105,7 @@ OM_uint32 gpm_accept_sec_context(OM_uint32 *minor_status,
     if (outbuf) {
         *output_token = *outbuf;
         free(outbuf);
+        outpuf = NULL;
```
Comment (simo5) [inline src/client/gpm_accept_sec_context.c:108]: -> outbuf

File: systemd/gssproxy.service.in [hunk 2]
```diff
@@ -6,6 +6,7 @@ Before=rpc-gssd.service
 [Service]
 ConfigurationDirectory=gssproxy
 StateDirectory=gssproxy gssproxy/clients gssproxy/rcache
+StateDirectoryMode=0700
```
Comment (simo5) [inline systemd/gssproxy.service.in:9]: Why this change ?

**Review by simo5 — CHANGES_REQUESTED** [review]
One broken change, and one question.

Changing the tests is a bit pedantic given all file descriptors get closed when the test completes anyway, but I am not against keeping static analysis tools quiet.
I would suggest though that you simply change your configuration of your static analysis tools to ignore the tests directory, for tests this is just churn and we should avoid it.

File: systemd/gssproxy.service.in (hunk 2 above)
Comment (simo5) [reply systemd/gssproxy.service.in:9]: Ok, I do not think there is any problem leaving this as 755 even in the spec file, given files are always created with restrictive permissions, but I do not see a huge deal in making it root only either ... it just makes the "run_as_user" setting difficult to use, because now you have to run around and change a bunch of directory permissions I guess ... but I never cared too much for tat setting.

Comment (jrisc) [reply systemd/gssproxy.service.in:9]: Since systemd overrides the cache directories according to the service unit file content, I guess that if the `User` setting is modified, it will set the owner of the cache directories accordingly. So it should not get in the way of non-root users to run the daemon.

The main motivation is to silent the rpminspect tests. Because gssproxy is started as part of the post-install script it modifies the directories mode immediately, which is causing installation tests to fail decause mode does not match the specfile anymore.

Comment (simo5) [reply systemd/gssproxy.service.in:9]: yeah, that is fine.

**Review by simo5 — APPROVED** [review]
LGTM
